staging/src/k8s.io/code-generator/generate-internal-groups.sh

```diff
@@ -47,15 +47,15 @@ EXT_APIS_PKG="$4"
 GROUPS_WITH_VERSIONS="$5"
 shift 5
 
-go install ${GOFLAGS:-} ./$(dirname "${0}")/cmd/{defaulter-gen,conversion-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
+go install ${GOFLAGS:-} ./"$(dirname "${0}")"/cmd/{defaulter-gen,conversion-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
```

Comment:
How to fix `go install ${GOFLAGS:-}` here ? According to https://github.com/koalaman/shellcheck/wiki/SC2086, need to quote `${GOFLAGS:-}`, but actually, it will failed by `go install ''` or `go install '-i -xxx'`.

Reply:
https://github.com/koalaman/shellcheck/wiki/SC2086#exceptions

this is the first situation, where we are building a command line, try `"${GOFLAGS[@]:-}"`

Reply:
@BenTheElder  already tried, same error with `go install ''` or `go install '-i -xxx'`. The problem caused by the quotation marks.

Reply:
Er right, because we're getting a literal empty string instead of a zero length value ... this one is fun -- `${GOFLAGS:+"${GOFLAGS}"}` 😅
If unset we'll get a null, and if set we'll get the quoted value.

Reply:
cool, but it cannot fix `go install '-i -xxx'`. 😔

Reply:
how are you testing this? `${GOFLAGS:+"${GOFLAGS[@]}"}` or perhaps we turn off the lint for this line which is otherwise benign anyhow? having trouble replicating your failures locally :/

Reply:
oh, this works `${GOFLAGS:+${GOFLAGS[@]}}`, both for empty and multiple parameters. 😀

Reply:
😅 thereee we go. now we just have to figure out that other one :P 🚒